File: frontend/assets/ts/core/openai/protocols.ts
/* SoAI - Cross-feature OpenAI capability descriptor protocols [frontend/assets/ts/core/openai/protocols.ts] */
// SPDX-License-Identifier: LicenseRef-SoAI-Source-1.0

interface CapabilityDescriptor {
    id: string;
    label: string;
    className: string;
    title?: string;
}

export type { CapabilityDescriptor };

File: backend/core/openai/protocols.py
"""SoAI - Core OpenAI protocol contracts [backend/core/openai/protocols.py]"""
# SPDX-License-Identifier: LicenseRef-SoAI-Source-1.0

from __future__ import annotations

from typing import Protocol

__all__ = ("PromptTokenCountStateProtocol",)


class PromptTokenCountStateProtocol(Protocol):
    total_tokens: int
    counted_characters: int
    capped: bool
    reason: str | None
